api/envoy/config/listener/v3/listener_components.proto

```diff
@@ -64,6 +64,18 @@ message Filter {
 // ``www.example.com``, then ``*.example.com``, then ``*.com``, then any filter
 // chain without ``server_names`` requirements).
 //
+// Yet another approach to reason about the filter chain matches.
```

Comment:
To me, the sieve explanation is less intuitive than an ordered walk through the criteria, but this is subjective. I'll defer to other @envoyproxy/api-shepherds on whether they think this adds value.